doc/sphinx/architecture.md

```diff
@@ -0,0 +1,131 @@
+# Overview
+This document describes the high-level architecture of *Mir*, including a diagram and a brief introduction of the directory structure.
+
+# Concepts
+## MirAL
+- The "Mir Abstraction Layer" (aka `mirAL` or `miral`) is a layer that makes the core Mir implementation easier for compositor-implementers to work with.
+- If one wishes to implement a compositor themselves, they will spend most of their time interacting with the `mirAL` API and only really touch the `mir` API when they require core concepts like a `mir::geometry::Rectangle`.
```

Comment:
`Rectangle` is from mircore, and that's very different from the mirserver APIs (that compositor writers should never touch.